### ISSUE TITLE
False positive in `require-mock-type-parameters` rule

### ISSUE THREAD

**Issue by FloEdelmann** [vitest-dev/eslint-plugin-vitest#654]
See https://github.com/FloEdelmann/vue-ts-types/pull/802/files#diff-b762127509fe80fbe15d39c56430d58dcdc3b2a8b267a4815e0ee4683698ccd7:

This is wrongly reported by the new [vitest/require-mock-type-parameters](https://github.com/vitest-dev/eslint-plugin-vitest/blob/main/docs/rules/require-mock-type-parameters.md) rule:

```ts
const validator3 = vi
  .fn<() => string | undefined>()
  .mockReturnValue('some error message');
```

These are both not reported:

```ts
const validator3 = vi
  .fn<() => string | undefined>(() => 'some error message');
```

```ts
const validator3 = vi
  .fn<() => string | undefined>();
```

**Comment by marekdedic** [comment]
Hmm, strange, let me take a look...

**Comment by marekdedic** [comment]
Thanks, I created a PR with a fix!
